app/controllers/rounds_controller.rb

```diff
@@ -48,7 +48,22 @@ def show
     @round = Round.find(params[:id])
     @game_content = GameContent.find(@round.game_content_id)
     @player_inputs = PlayerInput.where(round_id: @round.id)
-    # non_host_player_user_ids is an instace model method retrieving user_id of non-host players
+    # non_host_player_user_ids is an instance model method retrieving user_id of non-host players
     @non_host_player_user_ids = @instance.non_host_player_user_ids
   end
+
+  private
+
+  def generate_unique_question(instance, round)
+    game = Game.find(instance.game_id)
+    game_contents_array = game.game_contents
+
+    loop do
+      game_contents_array = Game.game_contents if game_contents_array.empty?
+      round.game_content_id = game_contents_array.sample.id
```

Comment:
Currently the `game_contents_array` will never be empty cause we're not `.pop`ing an instance out of it.

If you do `game_contents_array.shuffle!.pop.id`:
- `shuffle!` should re-order the elements inside the array
- `pop` should get the random element that ended up at being the last element of the array after the shuffle
- combination of the two creates a method like `sample!` which currently don't exist

Reply:
Thanks @spioupiou, one last thing 🙏🏼 

Currently the `game_contents_array` will never be empty cause we're not `.pop`ing an instance out of it.

If you do `game_contents_array.shuffle!.pop.id`:
- `shuffle!` should re-order the elements inside the array
- `pop` should get the random element that ended up at being the last element of the array after the shuffle
- combination of the two creates a method like `sample!` which currently don't exist